### ISSUE TITLE
Easy to get locked up when selecting unrestricted items

### ISSUE THREAD

**Issue by alethiophile** [ClaudeMetz/FactoryPlanner#222]
### Suggestion

The current interface for selecting unrestricted items has some land mines in it. Often when dealing with complex recipes it's not immediately clear which items should be unrestricted, and it requires some trial and error to see which ones make the result come out reasonable.

However, there are a few issues here. One is that the matrix solver may occasionally find a solution in which one or more of the provided recipes runs backward (negative coefficient). This is obviously just an error, and generally means you've suggested an unrestricted item that doesn't make sense; it would be good to detect this and disallow selections that lead to a negative number.

More seriously, if you get the "linearly dependent recipes" error, this removes your ability to select unrestricted items. Thus, the only way to fix it once you get that error is to remove one of the relevant recipes, add it back, and then make another selection. It also clears your existing selection, meaning that if you've got a working recipe and you get this error while modifying it, it can take some work and guessing even to get back to your working recipe.

You should be able to select unrestricted items like normal even after a "linearly dependent recipes" error, and making a new selection should rerun the solver again. This would allow trial and error in selecting items without fear of messing up your existing arrangement.

**Comment by ClaudeMetz** [comment]
Thanks, appreciate the well thought-out suggestion! Not showing things that lead to negative numbers is tricky, since it would require computing all the permutations to find that out, to my understanding. That is not really feasible, performance-wise unfortunately.

The other suggestions are great, I'll look into them when I get the chance :)

**Comment by jgoldshlag** [comment]
+1 for this one for me. Playing Nullius, I often choose the wrong unrestricted item, and having to delete a line and then re-add it to re-select the unrestricted items is a pain. I think the old UI used to let you re-choose the items, a simple button to clear unrestricted items and force me to re-select them again would be great

**Comment by ClaudeMetz** [comment]
This was long ago, but version ` 1.1.73` changed this UI. Does it still pose problems for you @jgoldshlag or is it okay now?

**Comment by jgoldshlag** [comment]
It is still an issue, there doesn't seem to be a way to back out of a selected unrestricted item when you choose wrongly. Still running into this in my Pyanadons run

**Comment by ClaudeMetz** [comment]
Okay thanks for letting me know, I'll look into it.

**Comment by puggan** [comment]
Still a problem in v2.0.9.

My work around is to switch solver to Traditional and back to Matrix.
It resets the "unrestricted" and get you out of "linearly dependent recipes".

One way to solve both without "computing all the permutations", is to just compute the one I clicked on, and if it fails, add it to a temporary blacklist and don't add it to the list (rollback), and only if it succeed add it to the list.
(reset the blacklist when a item is moved back to restricted, or when the plan is modified)

When a complete set is selected, use a different color for items that end up negative.

**Comment by ClaudeMetz** [comment]
Yeah I ran into this issue myself, I'll try to come up with a solution.

**Comment by ClaudeMetz** [comment]
Alrighty this is an ancient request, but it finally got solved in #586 for the next release.

To be clear, the part that got resolved was the 'getting stuck' one. It'll now allow you to unselect unrestricted items when there's too many. The other request about negative results is technically hard to solve, and I don't think we will, so I'll consider this whole request resolved.
